### PR TITLE
Add `spritePicker` custom input type

### PR DESCRIPTION
**Pull request by joshlory**
Sample usage:

```json
[{
  "blockText": "create a sprite named {SPRITE}",
  "func": "createSprite",
  "color": [72, 1, 0.8],
  "args": [{
    "name": "SPRITE",
    "customInput": "spritePicker"
  }]
}]
```

image

### CODE REVIEW

**Comment by joshlory** [comment]
[`customInput` dropdown for sprite name (variable picker)](https://trello.com/c/KOYnqQbY/132-custominput-dropdown-for-sprite-name-variable-picker)

File: apps/src/gamelab/blocks.js [hunk 1]
```diff
@@ -68,6 +68,34 @@ const customInputTypes = {
       return block.getTitleValue(arg.name);
     },
   },
+  spritePicker: {
+    addInput(block, input) {
+      Blockly.Variables.registerGetter(Blockly.BlockValueType.SPRITE, block.name);
```
Comment (balderdash) [inline apps/src/gamelab/blocks.js]: `registerGetter` marks a block as the one that you want to use in the corresponding autopopulated toolbox category, so it should just be called with `sprite_variables_get`, not every block that has a sprite variable dropdown.

Comment (balderdash) [reply apps/src/gamelab/blocks.js]: Hmm, it might be more accurately named `setGetter` or `setGetterForType`.

Comment (joshlory) [reply apps/src/gamelab/blocks.js]: Hmm, I'm not sure I fully understand the different options.  I removed this line, but I have the following bug.

1. Drag out a block that uses the `spritePicker` custom input.  Rename the variable:



2. Both variables show up in the "Sprite" toolbox category:


But only one of the variable names shows up in the `spritePicker` dropdown: 😕

Comment (balderdash) [reply apps/src/gamelab/blocks.js]: The sprite category shows you all the sprite variables you've used, along with one additional getter+setter that has a unique name (the normal variable category does this too).

It probably makes less sense to do this for sprite variables. https://github.com/code-dot-org/blockly/pull/105 provides a way to turn that off.

Comment (joshlory) [reply apps/src/gamelab/blocks.js]: Ah, got it.  So this is working as designed 👍

Comment (joshlory) [reply apps/src/gamelab/blocks.js]: I also get the same behavior if I choose "New variable..." in step (1) instead of "Rename variable..." — expected?

Comment (balderdash) [reply apps/src/gamelab/blocks.js]: Yep. There should be one extra pair of variables blocks even before you rename/create a new variable.

**Pull request merged**